src/components/Header/index.js

```diff
@@ -1,9 +1,57 @@
-import React from 'react';
+import React, { useState } from 'react';
 
-export default function Header() {
+import Container from '../Container';
+import ListLink from '../ListLink';
+import Logo from '../Logo';
+
+import headerStyles from './style.module.css';
+
+function Header() {
+  const [isToggled, setToggle] = useState(false);
+
+  const toggleMenu = () => {
+    setToggle(!isToggled);
+  };
   return (
```

Comment:
there is `@reach/router` in gatsby. i can get the location using `useLocation` hook in the component.

But i don't think it will be efficient to depend on hash changes and useEffect. the user might be on `#features`  section and then opens the menu and clicks on `#features` link again. Since hash does not change, the useEffect won't call the setToggle and the menu won't be closed. make sense?